File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.js
function solution(food) {
    console.log("준비한 음식", food);
    const center = "0";
    let left = "";
    let right = "";
    
    for(let i = 1; i < food.length; i++){
        const num = Math.floor(food[i] / 2)
        if(num > 0){
            const f = i.toString().repeat(num)
            left += f;
            right = f + right
        }
    }
    
    const result = left + center + right;
    
    console.log(result);
    
    return result;
    
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.py
def solution(food):
    dict = {}
    
    result = ""
    
    for i, m in enumerate(food):
        divisor = m // 2
        if(divisor != 0):
            dict[i] = str(i) * divisor
    
    for k in dict.keys():
        result += dict[k]
        
    result += '0'    
    
    for k in reversed(dict.keys()):
        result += dict[k]
    
    return result